src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -17,6 +17,17 @@ public static class ConfigurationExtensions
         private const string Secrets_File_Name = "secrets.json";
 
         /// <summary>
+        /// Adds the user secrets configuration source. Searches the assembly that contains type <typeparamref name="TStartup"/>
+        /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <typeparam name="TStartup">The startup class</typeparam>
+        /// <returns></returns>
+        public static IConfigurationBuilder AddUserSecrets<TStartup>(this IConfigurationBuilder configuration)
```

Comment:
Isn't TStartup a bit arbitrary? I get why it is nice but I feel guilty for lying :smile: I don't feel strongly though.